test/integration/serverTest.js

```diff
@@ -154,6 +157,150 @@ describe('Brave Sensor server', () => {
+      await firmwareAlert(radar_coreID, SENSOR_EVENT.STILLNESS)
+      expect(particle.callFunction).to.have.been.calledWith({
+        deviceId: 'particleCoreIdTest',
+        name: 'start',
+        argument: 'start',
+        auth: helpers.getEnvVar('PARTICLE_ACCESS_TOKEN'),
+      })
+    })
+
+    it('should not call startAlertSession if particleSirenID is not null', async () => {
+      const sessions = await db.getAllSessionsFromLocation(testLocation1Id)
+      expect(sessions.length).to.equal(0)
+    })
+  })
+
+  describe('POST request: alerts from firmware state machine for locations with a null Particle Siren Id', () => {
+    beforeEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+      const client = await clientFactory(db)
+      await db.createLocation(
+        testLocation1Id,
+        testLocation1PhoneNumber,
+        MOVEMENT_THRESHOLD,
+        STILLNESS_TIMER,
+        DURATION_TIMER,
+        1000,
+        INITIAL_TIMER,
+        ['+15005550006'],
+        '+15005550006',
+        ['+15005550006'],
+        1000,
+        'locationName',
+        door_coreID,
+        radar_coreID,
+        'XeThru',
+        'alertApiKey',
+        true,
+        true,
+        null,
+        client.id,
+      )
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+      sandbox.spy(particle, 'callFunction')
+      sandbox.spy(helpers, 'logError')
+    })
+    afterEach(async () => {
+      await redis.clearKeys()
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+      sandbox.restore()
+      helpers.log('\n')
+    })
+
+    it('should return 200 to a request with no headers', async () => {
+      const response = await chai.request(server).post('/api/sensorEvent').send({})
+      expect(response).to.have.status(200)
+    })
+
+    it('should log an error for a request with no headers', async () => {
+      await chai.request(server).post('/api/sensorEvent').send({})
+      expect(helpers.logError).to.have.been.called
+    })
+
+    it('should return 200 for a valid request', async () => {
+      const response = await firmwareAlert(radar_coreID, SENSOR_EVENT.STILLNESS)
+      expect(response).to.have.status(200)
+    })
+
+    it('should not call particle.callFunction if there sirenParticleId is null', async () => {
+      expect(particle.callFunction).to.have.not.been.called
+    })
+
+    it('should call startAlertSession if particleSirenId is null', async () => {
+      await firmwareAlert(radar_coreID, SENSOR_EVENT.STILLNESS)
+      const sessions = await db.getAllSessionsFromLocation(testLocation1Id)
+      const session = sessions[sessions.length - 1]
+      expect(session.alertType).to.equal(ALERT_TYPE.SENSOR_STILLNESS)
+    })
```

Comment:
This one's name doesn't quite match up with what you're testing either. 
I'd either change the name to something like `should create a session with the right alert type if particleSirenId is null`
Or change the test to spy on `braveAlerter.startAlertSession` and test that it is called with the expected parameters by using an expects something like `expect(braveAlerter.startAlertSession).to.be.calledWithExactly({....})`

Reply:
Updated the test name